mobile_verifier/src/coverage.rs

```diff
@@ -0,0 +1,721 @@
+    coverage_obj: file_store::coverage::CoverageObject,
+    validity: CoverageObjectValidity,
+}
+
+impl CoverageObject {
+    pub fn validate_coverage_objects<'a>(
+        auth_client: &'a AuthorizationClient,
+        coverage_objects: impl Stream<Item = CoverageObjectIngestReport> + 'a,
+    ) -> impl Stream<Item = anyhow::Result<Self>> + 'a {
+        coverage_objects.then(move |coverage_object_report| async move {
+            let validity = validate_coverage_object(&coverage_object_report, auth_client).await?;
+            Ok(CoverageObject {
+                coverage_obj: coverage_object_report.report,
+                validity,
+            })
+        })
+    }
+
+    pub async fn write(&self, coverage_objects: &FileSinkClient) -> file_store::Result {
+        coverage_objects
+            .write(
+                proto::CoverageObjectV1 {
+                    coverage_object: Some(proto::CoverageObjectReqV1 {
+                        pub_key: self.coverage_obj.pub_key.clone().into(),
+                        uuid: Vec::from(self.coverage_obj.uuid.into_bytes()),
+                        cbsd_id: self.coverage_obj.cbsd_id.clone(),
+                        coverage_claim_time: self
+                            .coverage_obj
+                            .coverage_claim_time
+                            .encode_timestamp(),
+                        indoor: self.coverage_obj.indoor,
+                        coverage: self
+                            .coverage_obj
+                            .coverage
+                            .clone()
+                            .into_iter()
+                            .map(Into::into)
+                            .collect(),
+                        signature: self.coverage_obj.signature.clone(),
+                    }),
+                    validity: self.validity as i32,
+                },
+                &[("validity", self.validity.as_str_name())],
+            )
+            .await?;
+        Ok(())
+    }
+
+    pub async fn save(self, transaction: &mut Transaction<'_, Postgres>) -> anyhow::Result<bool> {
+        // If the coverage object is not valid, do not save it
+        if self.validity != CoverageObjectValidity::Valid {
+            return Ok(false);
+        }
+
+        for hex in self.coverage_obj.coverage {
+            let location: u64 = hex.location.into();
+            sqlx::query(
+                r#"
+                INSERT INTO hex_coverage
+                  (uuid, hex, indoor, cbsd_id, signal_level, coverage_claim_time, inserted_at)
+                VALUES
+                  ($1, $2, $3, $4, $5, $6, $7)
+                "#,
+            )
+            .bind(self.coverage_obj.uuid)
+            .bind(location as i64)
+            .bind(self.coverage_obj.indoor)
+            .bind(&self.coverage_obj.cbsd_id)
+            .bind(SignalLevel::from(hex.signal_level))
+            .bind(self.coverage_obj.coverage_claim_time)
+            .bind(Utc::now())
```

Comment:
As per suggestion in the migration, if `inserted_at` uses default now(), you can eliminate this binding